src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,478 @@
+
+    return core_accn_meta
+
+
+def datasets_asm_reports(
+    sif_image: str, assembly_accessions: dict, download_directory: PathLike, batch_size: int
+) -> Dict:
+    """Obtain multiple assembly report JSONs in one or more querys to datasets,
+    i.e. make individual since accn query to datasets tool.
+
+    Args:
+        sif_image: Instance of Client.loaded singularity image.
+        assembly_accessions: Dict of core accessions.
+        download_directory: Dir path to store assembly report JSON files.
+        batch_size: Number of assembly accessions to batch submit to 'datasets'.
+
+    Returns:
+        Dictionary of core name and its associated assembly report
+    """
+
+    master_accn_list = list(assembly_accessions.values())
+    combined_asm_reports = {}
+
+    # Setting the number of combined accessions to query in a single call to datasets
+    list_split = list(range(0, len(master_accn_list), batch_size))  ## Note best to use >=10
+    accn_subsample = [master_accn_list[ind : ind + batch_size] for ind in list_split]
+
+    for accessions in accn_subsample:
+        datasets_command = ["datasets", "summary", "genome", "accession"] + accessions
+
+        # Make call to singularity datasets providing a multi accn query:
+        client_return = Client.execute(
+            image=sif_image, command=datasets_command, return_result=True, quiet=True
+        )
+
+        result = client_return["message"]
+
+        ## Test what result we have returned following execution of sif image and accession value
+        # Returned a str, i.e. no datasets result obtained exited with fatal error
+        if not isinstance(result, str):
+            raise ValueError("Result obtained from datasets is not the expected format 'string'")
+        if re.search("^FATAL", result):
+            logging.critical(f"Singularity image execution failed! -> '{result.strip()}'")
+            sys.exit(1)
+        # Returned a list, i.e. datasets returned a result to client.execute
+
+        tmp_asm_dict = json.loads(result)
+        if tmp_asm_dict["total_count"] >= 1:
+            logging.info(f"Asm report obtained for accession(s) [{accessions}]")
+
+            batch_reports_json = tmp_asm_dict["reports"]
+            for assembly_report in batch_reports_json:
+                accession = assembly_report["accession"]
+                asm_json_outfile = f"{download_directory}/{accession}.asm_report.json"
+                print_json(Path(asm_json_outfile), assembly_report)
+                # Save assembly report into master core<>report dict
+                for core, accession_core in assembly_accessions.items():
+                    if accession == accession_core:
+                        combined_asm_reports[core] = assembly_report
+        else:
+            logging.warning(f"No assembly report found for accession(s) {accessions}. Exiting !")
+            sys.exit(0)
```

Comment:
```suggestion
```
I would rather report an empty dictionary and have a check in the TSV file creation to not create the file if no report was returned.